wrapper/src/main/java/software/amazon/jdbc/plugin/bluegreen/routing/BaseConnectRouting.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.bluegreen.routing;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.logging.Logger;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import software.amazon.jdbc.ConnectionPlugin;
+import software.amazon.jdbc.HostSpec;
+import software.amazon.jdbc.JdbcCallable;
+import software.amazon.jdbc.PluginService;
+import software.amazon.jdbc.plugin.bluegreen.BlueGreenConnectionPlugin;
+import software.amazon.jdbc.plugin.bluegreen.BlueGreenRole;
+
+public abstract class BaseConnectRouting extends BaseRouting implements ConnectRouting {
+
+  private static final Logger LOGGER = Logger.getLogger(BaseConnectRouting.class.getName());
+
+  protected final String hostAndPort; // if value is provided then host is mandatory and port is optional.
+  protected final BlueGreenRole role;
+
+  public BaseConnectRouting(@Nullable String hostAndPort, @Nullable BlueGreenRole role) {
+    this.hostAndPort = hostAndPort == null ? null : hostAndPort.toLowerCase();
+    this.role = role;
+  }
```

Comment:
Can we move these and the `toString` override to BaseRouting? Both this class `BaseExecuteRouting` have the same implementation.